datafusion/sql/src/statement.rs

```diff
@@ -505,6 +533,192 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
+            let col_name: &Ident = assign
+                .id
+                .iter()
+                .last()
+                .ok_or(DataFusionError::Plan("Empty column id".to_string()))?;
+            let col_name = col_name.value.as_str();
+            values.push((col_name.to_string(), assign.value.clone()));
+            let (_, field) =
+                table_schema
+                    .column_with_name(col_name)
+                    .ok_or(DataFusionError::Plan(format!(
+                        "Column not found: {col_name}"
+                    )))?;
+            let field = DFField::from(field.clone());
+            fields.push(field);
+        }
+
+        // Build scan
+        let from = from.unwrap_or(table);
+        let scan = self.plan_from_tables(vec![from], &mut planner_context)?;
+
+        // Filter
+        let source = match predicate_expr {
+            None => scan,
+            Some(predicate_expr) => {
+                let plan_schema = scan.schema();
+                let filter_expr =
+                    self.sql_to_expr(predicate_expr, plan_schema, &mut planner_context)?;
+                let schema = Arc::new(plan_schema.clone());
+                let mut using_columns = HashSet::new();
+                expr_to_columns(&filter_expr, &mut using_columns)?;
+                for use_col in using_columns.iter() {
+                    let col_name = use_col.name.clone();
+                    let rel = use_col.relation.as_ref().cloned();
+                    let field = plan_schema
+                        .field_with_name(rel.as_deref(), use_col.name.as_str())?
+                        .clone();
+                    fields.push(field.clone());
+                    values.push((
+                        col_name.clone(),
+                        ast::Expr::Identifier(ast::Ident::from(col_name.as_str())),
+                    ));
+                }
+                let filter_expr = normalize_col_with_schemas(
+                    filter_expr,
+                    &[&schema],
+                    &[using_columns],
+                )?;
+                LogicalPlan::Filter(Filter::try_new(filter_expr, Arc::new(scan))?)
+            }
+        };
+
+        // Projection
+        let proj_schema = DFSchema::new_with_metadata(fields, HashMap::new())?;
+        let mut exprs = vec![];
+        for (col_name, expr) in values.into_iter() {
+            let expr = self.sql_to_expr(expr, &proj_schema, &mut planner_context)?;
+            let expr = expr.alias(col_name);
+            exprs.push(expr);
+        }
+        let source = project(source, exprs)?;
+
+        let plan = LogicalPlan::Write(WriteRel {
+            table_name: table_ref,
+            table_schema: proj_schema.into(),
+            op: WriteOp::Update,
+            input: Arc::new(source),
+        });
+        Ok(plan)
+    }
+
+    fn insert_to_plan(
+        &self,
+        #[cfg_attr(feature = "visitor", visit(with = "visit_relation"))]
```

Comment:
I think this cfg stuff is out of sqlparser, not datafusion right?

Reply:
I can't find where that came from, but I removed it.